test/functional/spec-runner/index.js

```diff
@@ -125,20 +125,19 @@ function generateTopologyTests(testSuites, testContext, filter) {
       describe(suiteName, {
         metadata: { requires },
         test: function () {
-          beforeEach(() => prepareDatabaseForSuite(testSuite, testContext));
-          afterEach(() => testContext.cleanupAfterSuite());
+          // beforeEach(() => );
```

Comment:
I was getting leaked session errors for the tests that were skipped using `this.skip()`, which seemed to be because `beforeEach`/`afterEach` run even on tests skipped at runtime. Moving the hook behavior into the test resolved the issue, and I don't see any advantage of using the beforeEach/afterEach hooks since all of the tests are generated with the same code. 

I'm not sure if this just popped up or has been happening for a while. It wasn't a fatal error, so it wouldn't have caused any failures in evergreen.